test/interactions/interactionTests.ts

```diff
@@ -0,0 +1,94 @@
+      var callback = () => callbackCalled = true;
+      clickInteraction.onClick(callback);
+
+      clickInteraction.attachTo(component);
+
+      TestMethods.triggerFakeMouseEvent("mousedown", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      TestMethods.triggerFakeMouseEvent("mouseup", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      assert.isTrue(callbackCalled, "callback called on clicking Component (mouse)");
+
+      callbackCalled = false;
+      clickInteraction.detachFrom(component);
+
+      TestMethods.triggerFakeMouseEvent("mousedown", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      TestMethods.triggerFakeMouseEvent("mouseup", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      assert.isFalse(callbackCalled, "callback was removed from component and should not be called");
+
+      svg.remove();
+    });
+
+    it("interactions are reusable", () => {
+      var svg1 = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+      var svg2 = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+      var component1 = new Plottable.Component();
+      var component2 = new Plottable.Component();
+
+      component1.renderTo(svg1);
+      component2.renderTo(svg2);
+
+      var clickInteraction = new Plottable.Interactions.Click();
+
+      var callbackCalled: Boolean;
+      var callback = () => callbackCalled = true;
+      clickInteraction.onClick(callback);
+
+      clickInteraction.attachTo(component1);
+
+      callbackCalled = false;
+      TestMethods.triggerFakeMouseEvent("mousedown", component1.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      TestMethods.triggerFakeMouseEvent("mouseup", component1.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      assert.isTrue(callbackCalled, "Round 1 callback called for component 1");
+
+      callbackCalled = false;
+      TestMethods.triggerFakeMouseEvent("mousedown", component2.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      TestMethods.triggerFakeMouseEvent("mouseup", component2.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      assert.isFalse(callbackCalled, "Round 1 callback not called for component 2");
+
+      clickInteraction.detachFrom(component1);
+      clickInteraction.attachTo(component2);
+
+      callbackCalled = false;
+      TestMethods.triggerFakeMouseEvent("mousedown", component1.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      TestMethods.triggerFakeMouseEvent("mouseup", component1.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      assert.isFalse(callbackCalled, "Round 2 callback not called for component 1");
+
+      callbackCalled = false;
+      TestMethods.triggerFakeMouseEvent("mousedown", component2.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      TestMethods.triggerFakeMouseEvent("mouseup", component2.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      assert.isTrue(callbackCalled, "Round 2 callback called for component 2");
+
+      // no detach, as this shorthand should also work
+      clickInteraction.attachTo(component1);
```

Comment:
I'd like to avoid writing extra tests to test shorthand for the same thing. Tests are already running for 6 seconds (or more on SauceLabs). I'd also want to make it obvious that the detach was intentional. Not sure if it is without the comment.